### ISSUE TITLE
matlab2netCDF access to nested variables

### ISSUE THREAD

**Issue by atdyer** [FRF-Remote-Sensing/DuneLidar#14]
@Bobfrat 

Take a look at this [sample dataset](https://github.com/FRF-Remote-Sensing/DuneLidar/blob/fdif/FDIF/samples/linescan_gauges.mat) we've got set up for testing, as well as what I've got so far for the [template](https://github.com/FRF-Remote-Sensing/DuneLidar/blob/fdif/FDIF/templates/linescan_wave_gauge.yml). Most of our data structures around going to have multiple levels of nesting, e.g. a wave gauge will look like:
- location (struct)
  - latitude (double)
  - longitude (double)
  - xstd (double)
  - ystd (double)
- FFT (struct)
  - amplitude (1xn double)
  - frequency (1xn double)
  - sig (double)
  - sin (double)
- ts (struct)
  - time (1xm double)
  - z (1xm double)
- hs (double)
- wl (double)
- etc.

I can't get the matlab2netcdf script to work, and it seems to be because we've got variables nested within structs. Is this correct? Or am I defining something incorrectly/using the script wrong? Is there some other way I can define the variables in the template file so that it can be discovered by the script? We'd prefer not to have to change the way the data is organized if at all possible.

Here's the command I used to run the script:
```matlab
matlab2netCDF(lineWaveGaugeData.gauge{1}, 'metadata/linescan_wave_gauge_metadata.yml', 'templates/linescan_wave_gauge.yml', 1, '/home/tristan/Desktop/test.nc');
```

**Comment by Bobfrat** [comment]
@atdyer Currently the script is designed to work on a single matlab structure without nested structures. I think with a little thought we can modify it to work with your data structure. I'll take a look early next week.

**Comment by Bobfrat** [comment]
@atdyer can you send me the link to ```metadata/linescan_wave_gauge_metadata.yml``` as well?

**Comment by atdyer** [comment]
@Bobfrat [metadata](https://github.com/FRF-Remote-Sensing/DuneLidar/blob/fdif/FDIF/metadata/linescan_wave_gauge_metadata.yml) - not much has changed here from the template you sent, we'll be updating this one early next week.

**Comment by Bobfrat** [comment]
@atdyer What is the time stamp for this sample data? Is that in the data structure?

**Comment by atdyer** [comment]
@Bobfrat ts.time will have the time data in seconds since epoch (it's currently just dummy data, we're going to be converting it from the scanner's internal time to epoch).

**Comment by Bobfrat** [comment]
@atdyer Is ts.time going to be different than the time variable for the computed wave heights and water levels? For instance, I see 1000 time stamps in ts.time but one hs value. Can we add another time variable on the 'root' level of that data structure to denote the record time stamp? Are these files always going to contain just one record for wave height and water level?

**Comment by atdyer** [comment]
Right, the ts.z values that go with ts.time are instantaneous water levels, and the values hs and wl are significant wave height and average water level over the whole (30 minute) collect period.

@KateBrodie @hokiespurs can you weigh in on this? We'll need to include a timestamp to go with significant wave height, average water level, etc. I'm guessing we'll just use the start of the collect period?

**Comment by Bobfrat** [comment]
Might want to see what Kent does, he does something similar with the wave and current data. Not sure if it's start, end, or midpoint of the sampling period.

**Comment by Bobfrat** [comment]
@atdyer I just realized that script assumes that time is in matlab datenum's so there is a function provided in that zipped folder to convert matlab time to unix time.

**Comment by KateBrodie** [comment]
@atdyer @Bobfrat

Kent uses start of sampling period for all the FRF oceanographic data.
Let's conform.

**Comment by KateBrodie** [comment]
@atdyer can this issue be closed?
